astropy/config/paths.py

```diff
@@ -185,6 +184,10 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> None:
 
         return wrapper
 
+    @classmethod
```

Comment:
I'm intentionally avoiding `astropy.utils.classproperty` because this module is so low in the stack that I don't think it's wise to make it depend on other parts of astropy